File: src/main/scala/com/jjw/spark/test1/CacheTest.scala
package com.jjw.spark.test1

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext

object CacheTest {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf()
    conf.setMaster("local").setAppName("cacheTest")
    val sc = new SparkContext(conf)
    var lines = sc.textFile("./NASA_access_log_Aug95")
    lines = lines.cache()
    val startTime = System.currentTimeMillis()
    val count = lines.count()
    val endTime = System.currentTimeMillis()
    println("共"+count +"条数据，用时"+(endTime-startTime)+"ms")
    
    val startTime1 = System.currentTimeMillis()
    val count1 = lines.count()
    val endTime1 = System.currentTimeMillis()
    println("共"+count1 +"条数据，用时"+(endTime1-startTime1)+"ms")
    
    sc.stop()
  }
}

File: src/main/java/com/jjw/sparkCore/others/CacheTest.java
package com.jjw.sparkCore.others;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.storage.StorageLevel;
public class CacheTest{
	public static void main(String[] args) {
		SparkConf conf = new SparkConf();
		conf/*.setMaster("local")*/.setAppName("CacheTest");
		JavaSparkContext jsc = new JavaSparkContext(conf);
		JavaRDD<String> lines = jsc.textFile("hdfs://node1:9000/test/NASA_access_log_Aug95");
//		lines = lines.cache();
//		lines = lines.persist(StorageLevel.MEMORY_ONLY_2());
		lines = lines.persist(new StorageLevel(false, true, false, true, 3));
		long startTime = System.currentTimeMillis();
		long count = lines.count();
		long endTime = System.currentTimeMillis();
		System.out.println("共"+count+ "条数据，"+"初始化时间+cache时间+计算时间="+ (endTime-startTime));
		/*
		long countStartTime = System.currentTimeMillis();
		long countrResult = lines.count();
		long countEndTime = System.currentTimeMillis();
		System.out.println("共"+countrResult+ "条数据，"+"计算时间="+ (countEndTime-countStartTime));
		lines.unpersist();
		long countStartTime2 = System.currentTimeMillis();
		long countrResult2 = lines.count();
		long countEndTime2 = System.currentTimeMillis();
		System.out.println("共"+countrResult2+ "条数据，"+"计算时间="+ (countEndTime2-countStartTime2));*/
		
		
//		lines.unpersist();
		while(true){
			
		}
		
//		jsc.stop();
	}
}
